app/src/org/commcare/connect/PersonalIdManager.java

```diff
@@ -620,12 +640,16 @@ public static AuthInfo.TokenAuth getHqTokenIfLinked(String username) throws Toke
         }
 
         String seatedAppId = CommCareApplication.instance().getCurrentApp().getUniqueId();
-        ConnectLinkedAppRecord appRecord = ConnectAppDatabaseUtil.getConnectLinkedAppRecord(manager.parentActivity, seatedAppId, username);
-        if(appRecord == null) {
+
+        if (!manager.isPersonalIdLinkedApp(seatedAppId, username)) {
```

Comment:
what is the benefit of adding `isSeatedAppLinkedToPersonalId` if we are not going to use it here!